plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/transition/MapboxViewportCameraTransition.kt

```diff
@@ -0,0 +1,266 @@
+package com.mapbox.maps.plugin.viewport.transition
+
+import android.animation.Animator
+import android.animation.AnimatorSet
+import android.view.animation.Interpolator
+import androidx.core.view.animation.PathInterpolatorCompat
+import com.mapbox.geojson.Point
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.plugin.animation.CameraAnimationsPlugin
+import com.mapbox.maps.plugin.animation.CameraAnimatorOptions.Companion.cameraAnimatorOptions
+import com.mapbox.maps.plugin.delegates.MapCameraManagerDelegate
+import com.mapbox.maps.plugin.viewport.ViewportPluginImpl.Companion.VIEWPORT_CAMERA_OWNER
+import kotlin.math.abs
+
+/**
+ * Helper class that provides default implementation of [ViewportCameraTransition] generators.
+ */
+internal class MapboxViewportCameraTransition(
+  private val cameraManager: MapCameraManagerDelegate,
+  private val cameraPlugin: CameraAnimationsPlugin
+) : ViewportCameraTransition {
+  override fun transitionFromLowZoomToHighZoom(
+    cameraOptions: CameraOptions,
+    transitionOptions: ViewportCameraTransitionOptions
+  ): AnimatorSet {
+    val animators = mutableListOf<Animator>()
+    val currentMapCameraState = cameraManager.cameraState
+
+    var centerDuration = 0L
+    cameraOptions.center?.let { center ->
+      val screenDistanceFromMapCenterToLocation = screenDistanceFromMapCenterToTarget(
+        cameraManager = cameraManager,
+        currentCenter = currentMapCameraState.center,
+        targetCenter = center
+      )
+
+      val centerAnimationRate = 500L
```

Comment:
There are so many magic numbers here.

Reply:
It is a timings for animation. If you would like to proceed the animation faster or slowly please adjust constants.

Reply:
let's iterate in follow-up PRs.